Gems/PhysX/Code/Source/Pipeline/MeshExporter.cpp

```diff
@@ -318,18 +320,28 @@ namespace PhysX
 
                             materialName = localSourceSceneMaterialsList[materialId];
 
-                            // Keep using the first material when it has to be limited to one.
-                            if (limitToOneMaterial &&
-                                assetMaterialData.m_sourceSceneMaterialNames.size() == 1 &&
-                                assetMaterialData.m_sourceSceneMaterialNames[0] != materialName)
+                            // Use the first material found in the mesh when it has to be limited to one.
+                            if (limitToOneMaterial)
                             {
-                                materialName = assetMaterialData.m_sourceSceneMaterialNames[0];
+                                nodeMaterials.insert(materialName);
+                                if (firstMaterial.empty())
```

Comment:
Q: where is `firstMaterial` being set/initialized? i cannot see how `if (firstMaterial.empty())` ever fails.

Reply:
It's an AZStd::string, that must surely initialize to empty.

Reply:
@greerdv is correct, the AZStd;:string is empty by default.